mzLib/Proteomics/RetentionTimePrediction/Chronologer/Chronologer.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.IO;
+using TorchSharp;
+using TorchSharp.Modules;
+
+namespace Proteomics.RetentionTimePrediction.Chronologer
+{
+    /// <summary>
+    /// Chronologer is a deep learning model for highly accurate prediction of peptide C18 retention times (reported in % ACN).
+    /// Chronologer was trained on a new large harmonized database of > 2.6 million retention time observations
+    /// (2.25M unique peptides) constructed from 11 community datasets
+    /// and natively supports prediction of 17 different modification types.
+    /// With only a few observations of a new modification type (> 10 peptides),
+    /// Chronologer can be easily re-trained to predict up to 10 user supplied modifications.
+    ///
+    /// Damien Beau Wilburn, Ariana E. Shannon, Vic Spicer, Alicia L. Richards, Darien Yeung, Danielle L. Swaney, Oleg V. Krokhin, Brian C. Searle
+    /// bioRxiv 2023.05.30.542978; doi: https://doi.org/10.1101/2023.05.30.542978
+    /// 
+    /// https://github.com/searlelab/chronologer
+    ///
+    /// Licensed under Apache License 2.0
+    /// 
+    /// </summary>
+    internal class Chronologer : torch.nn.Module<torch.Tensor, torch.Tensor>
+    {
+        internal Chronologer() : this(Path.Combine(AppDomain.CurrentDomain.BaseDirectory,
+            "RetentionTimePrediction",
+            "Chronologer", "Chronologer_20220601193755_TorchSharp.dat"))
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the Chronologer model class with pre-trained weights from the paper
+        /// Deep learning from harmonized peptide libraries enables retention time prediction of diverse post
+        /// translational modifications paper (https://github.com/searlelab/chronologer).
+        /// Eval mode is set to true and training mode is set to false by default.
+        ///
+        /// Please use .Predict() for using the model, not .forward(). 
+        /// </summary>
+        /// <param name="weightsPath"></param>
+        /// <param name="evalMode"></param>
+        internal Chronologer(string weightsPath, bool evalMode = true) : base(nameof(Chronologer))
+        {
+            RegisterComponents();
+
+            LoadWeights(weightsPath);//loads weights from the file
+
+            if (evalMode)
+            {
+                eval();
+                train(false);
+            }
+        }
+
+        /// <summary>
+        /// Do not use for inferring. Use .Predict() instead.
+        /// </summary>
```

Comment:
but what IS this used for.... add comment

Reply:
explanation on why forward() is not used when predicting outside the training method. https://stackoverflow.com/questions/58508190/in-pytorch-what-is-the-difference-between-forward-and-an-ordinary-method 

it has a more technical explanation that I don't fully understand (yet), but it is recommended. Will be adding a comment in the summary section